group-coordinator/src/main/java/org/apache/kafka/coordinator/group/ClassicGroupMetadataManager.java

```diff
@@ -0,0 +1,2500 @@
+import static org.apache.kafka.coordinator.group.metrics.GroupCoordinatorMetrics.CONSUMER_GROUP_REBALANCES_SENSOR_NAME;
+
+public class ClassicGroupMetadataManager {
+
+    private final GroupStore groupStore;
+    private final GroupMetaManagerHelper helper;
+
+    /**
+     * The maximum number of members allowed in a single classic group.
+     */
+    private final int classicGroupMaxSize;
+
+    /**
+     * Initial rebalance delay for members joining a classic group.
+     */
+    private final int classicGroupInitialRebalanceDelayMs;
+
+    /**
+     * The metadata refresh interval.
+     */
+    private final int consumerGroupMetadataRefreshIntervalMs;
+
+    /**
+     * The timeout used to wait for a new member in milliseconds.
+     */
+    private final int classicGroupNewMemberJoinTimeoutMs;
+
+    /**
+     * The maximum number of members allowed in a single consumer group.
+     */
+    private final int consumerGroupMaxSize;
+
+    /**
+     * The coordinator metrics.
+     */
+    private final GroupCoordinatorMetricsShard metrics;
+
+    /**
+     * The system timer.
+     */
+    private final CoordinatorTimer<Void, CoordinatorRecord> timer;
+
+    /**
+     * The system time.
+     */
+    private final Time time;
+
+    /**
+     * The log context.
+     */
+    private final LogContext logContext;
+
+    /**
+     * The logger.
+     */
+    private final Logger log;
+
+    /**
+     * The supported consumer group partition assignors keyed by their name.
+     */
+    private final Map<String, ConsumerGroupPartitionAssignor> consumerGroupAssignors;
+
+    /**
+     * The default consumer group assignor used.
+     */
+    private final ConsumerGroupPartitionAssignor defaultConsumerGroupAssignor;
+
+    /**
+     * The config indicating whether group protocol upgrade/downgrade is allowed.
+     */
+    private final ConsumerGroupMigrationPolicy consumerGroupMigrationPolicy;
+
+
+    /**
+     * An empty result returned to the state machine. This means that
+     * there are no records to append to the log.
+     *
+     * Package private for testing.
+     */
+    static final CoordinatorResult<Void, CoordinatorRecord> EMPTY_RESULT =
+        new CoordinatorResult<>(Collections.emptyList(), CompletableFuture.completedFuture(null), false);
+
+
+    public ClassicGroupMetadataManager(
+        GroupStore groupStore,
+        GroupCoordinatorMetricsShard metrics,
+        int classicGroupMaxSize,
+        int classicGroupInitialRebalanceDelayMs,
+        int consumerGroupMetadataRefreshIntervalMs,
+        int classicGroupNewMemberJoinTimeoutMs,
+        int consumerGroupMaxSize,
+        CoordinatorTimer<Void, CoordinatorRecord> timer,
+        Time time,
+        LogContext logContext,
+        List<ConsumerGroupPartitionAssignor> consumerGroupAssignors,
+        ConsumerGroupMigrationPolicy consumerGroupMigrationPolicy) {
```

Comment:
I think this would be a bit more legible with the line break after "policy" so the parentheses is at the state of the next line. Then the argument declarations will not blend into the assignments in the method body. There are a few other instances in this PR which would also benefit from the same formatting tweak.